server/src/controllers/auth.js

```diff
@@ -0,0 +1,45 @@
+import axios from 'axios';
+import User, { validateUser } from '../models/user';
+
+export const checkUser = async (userData) => {
+  const { error } = validateUser(userData);
+  if (error) return error.details[0].message;
+
+  let user = await User.findOne({ email: userData.email });
+  if (user) {
+    return true;
+  }
+  user = new User(userData);
+  user = await user.save();
+  if (user) {
+    return true;
+  }
+  return false;
```

Comment:
Last couple lines could be.

return user.create(userData, (err, user) => {
  if (err) { return false; }
  return true;
})